cpp/src/slice2java/Gen.cpp

```diff
@@ -1840,22 +1855,24 @@ Slice::Gen::TypesVisitor::visitClassDefStart(const ClassDefPtr& p)
                 baseDataMembers = baseClass->allDataMembers();
             }
 
-            if (hasRequiredMembers && hasOptionalMembers)
+            if (!requiredMembers.empty() && !optionalMembers.empty())
             {
                 //
                 // Generate a constructor accepting parameters for just the required members.
                 //
-                out << sp << nl << "public " << name << spar;
+                out << sp;
+                out << nl << "/**";
+                out << nl << " * Constructs a new {@code " << name << "} with required fields initialized.";
```

Comment:
I'd rather avoid the term `required`. What about:
```text
Creates a new NAME with values for all fields not marked optional in the Slice definition for NAME. 
```

Note that this constructor is unique to Java. I wish we did not generate it, however, dropping it would most likely break some existing code for no good reason.